components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx

```diff
@@ -365,12 +365,13 @@ const EditVisibleAssetsModal = ({ onClose }: Props) => {
     toggleShowAddCustomToken()
   }
 
-  const onRemoveAsset = (token: BraveWallet.BlockchainToken) => {
-    const newUserList = updatedTokensList.filter((t) => t.contractAddress.toLowerCase() !== token.contractAddress.toLowerCase())
+  const onRemoveAsset = React.useCallback((token: BraveWallet.BlockchainToken) => {
+    const filterFn = (t: BraveWallet.BlockchainToken) => !(t.contractAddress.toLowerCase() === token.contractAddress.toLowerCase() && t.tokenId === token.tokenId)
+    const newUserList = updatedTokensList.filter(filterFn)
     setUpdatedTokensList(newUserList)
-    const newFilteredTokenList = filteredTokenList.filter((t) => t.contractAddress.toLowerCase() !== token.contractAddress.toLowerCase())
+    const newFilteredTokenList = filteredTokenList.filter(filterFn)
     setFilteredTokenList(newFilteredTokenList)
```

Comment:
Refactoring opportunity: filteredTokenList should be computed via memo that uses the assetsList & search text (saved as state)

Reply:
@josheleonard Following our conversation about this, I created an issue to get this done: https://github.com/brave/brave-browser/issues/24408